### PR TITLE
cilium: Add CLI to introspect IP <-> Identity cache

### PR DESCRIPTION
**Pull request by joestringer**
The flow of IP <-> Identity mappings into the IPIdentityCache follow the
path of:

1) K8s watch / local poll / generation of the mapping
2) pkg/ipcache:IPIdentityCache
3) pkg/bpf/map userspace cache of map entries being pushed into BPF map
4) Actual BPF map

We can currently introspect some of (1) depending on the source; (3)
always; and (4) depending on the kernel version in use. (2) cannot be
introspected.

This commit adds support for introspecting (2) to assist in
troubleshooting issues related to the propagation of IP to Identity
mappings.

Example output:

```
# cilium ip list
IP                 IDENTITY   SOURCE
0.0.0.0/0          2
10.29.2.92/32      51185      k8s
10.29.25.191/32    1
10.29.27.15/32     4
10.29.27.93/32     104        k8s
10.29.74.136/32    30909      k8s
10.29.78.12/32     16083      k8s
10.29.89.252/32    20623      k8s
10.29.136.128/32   3157       k8s
10.29.177.163/32   19933      k8s
10.29.194.215/32   22010      k8s
10.29.209.106/32   59801      k8s
10.29.224.16/32    589        k8s
192.168.1.29/32    1
```

Related:  #11517

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: bugtool/cmd/configuration.go [hunk 1]
```diff
@@ -252,6 +252,7 @@ func copyCiliumInfoCommands(cmdDir string, k8sPods []string) []string {
 		"cilium bpf ipcache list",
 		"cilium bpf policy get --all --numeric",
 		"cilium bpf sha list",
+		"cilium ip list",
```
Comment (joestringer) [inline bugtool/cmd/configuration.go]: nit: We could probably do `cilium ip list --verbose` and/or `cilium ip list -o json` here for more info.

**Review by christarazi — APPROVED** [review]
Nice! LGTM

**Comment by joestringer** [comment]
test-me-please

**Comment by tklauser** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-me-please

**Review by pchaigno — APPROVED** [review]

**Review by pchaigno — CHANGES_REQUESTED** [review]
Actually, aren't the two generated files `Documentation/cmdref/cilium_ip.md` and `Documentation/cmdref/cilium_ip_list.md` missing from this PR?

**Comment by tklauser** [comment]
test-me-please

**Comment by joestringer** [comment]
> Actually, aren't the two generated files Documentation/cmdref/cilium_ip.md and Documentation/cmdref/cilium_ip_list.md missing from this PR?

Good catch.

**Comment by joestringer** [comment]
test-me-please

**Review by pchaigno — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Comment by pchaigno** [comment]
> > Actually, aren't the two generated files Documentation/cmdref/cilium_ip.md and Documentation/cmdref/cilium_ip_list.md missing from this PR?
> 
> Good catch. I fixed this up.

It's kinda of weird that the documentation tests didn't catch this. @errordeveloper Is that expected?

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
> It's kinda of weird that the documentation tests didn't catch this. @errordeveloper Is that expected?

I agree. Note that the build *did* complain about cmdref, but when I responded, I regenerated the cmdref and only added the diff into my commit. This type of failure is easy to miss in local testing since the file is still there, but I'd expect the CI side to pick up that there's a missing file or unclean git or broken links and complain there.

**Comment by joestringer** [comment]
Nominating for v1.7 backport to aid debugging.

**Comment by joestringer** [comment]
test-me-please

**Comment by tklauser** [comment]
Looks like the runtime tests failed because the `verbose` flag is not in the cmdref docs:

```
09:39:21      runtime: diff --git a/Documentation/cmdref/cilium_ip_list.md b/Documentation/cmdref/cilium_ip_list.md
09:39:21      runtime: index 45b1652d7..ab2934016 100644
09:39:21      runtime: --- a/Documentation/cmdref/cilium_ip_list.md
09:39:21      runtime: +++ b/Documentation/cmdref/cilium_ip_list.md
09:39:21      runtime: @@ -17,6 +17,7 @@ cilium ip list [flags]
09:39:21      runtime:  ```
09:39:21      runtime:    -h, --help            help for list
09:39:21      runtime:    -o, --output string   json| jsonpath='{}'
09:39:21      runtime: +  -v, --verbose         Print all fields of ipcache
09:39:21      runtime:  ```
09:39:21      runtime:  
09:39:21      runtime:  ### Options inherited from parent commands
09:39:21      runtime: HINT: to fix this, run 'make -C Documentation update-cmdref ; git commit Documentation/cmdref --message "docs: Update cmdref"'
```

**Comment by joestringer** [comment]
test-me-please

**Comment by errordeveloper** [comment]
I've opened an issue #11619 regarding CI checks.

**Pull request merged**